lib/kupong_integration/service.rb

```diff
@@ -0,0 +1,112 @@
+require 'json'
+
+module KupongIntegration  
+  class Service
+    DEFAULT_API_URL = 'https://api.kupong.se/v1.5/coupons'.freeze
+    SEND_PATH = '/send'.freeze
+    
+    PLUS = '+'.freeze
+    CODE = '46'.freeze
+    COUNTRY_CODE = (PLUS + CODE).freeze
+    EMPTY_STRING = ''.freeze
+    
+    DEFAULT_SETTINGS = {}.freeze
+    DEFAULT_API_PATH = ''.freeze
+    
+    SETTINGS_ATTRIBUTES = %i(
+      authorization
+      coupon_id
+    )
+    
+    SUCCESS_CODE = 200
+    CREATED_CODE = 201
+    
+    SANITIZE = '+ -'.freeze
+    
+    attr_reader :settings, :msisdn, :timestamp
+    
+    def self.config(api_url: nil)
+      @@api_url = api_url || DEFAULT_API_URL
+    end
+      
+    def initialize(settings: DEFAULT_SETTINGS, phone:)
+      @settings  = settings
+      @msisdn    = sanitize(phone)
+      @timestamp = Time.now.to_i.to_s
+    end
+    
+    def call 
+      response = create_coupon
+      send_coupon(response)
+    end
+    
+    private
+    
+    def api_url 
+      @@api_url ||= DEFAULT_API_URL
+    end
+    
+    def create_coupon
+      api_call(payload: create_payload)
+    end
+    
+    def send_coupon(response)
+      coupon_code = JSON.parse(response)['couponCode']
```

Comment:
Any danger of not being able to parse the response? do we need to rescue from this case?